### PR TITLE
Windows App Updates

### PR DESCRIPTION
**Pull request by nibanks**
General clean up with a bit of redesign to better use the OpenThread API with Visual C++ (C++ /CX). Also adds a readme for the App, with a few pictures.

### CODE REVIEW

File: examples/apps/windows/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# OpenThread App for Windows #
+
+This sample app provides an example of how to interface with the OpenThread API in a
+[Universal Windows App](https://developer.microsoft.com/en-us/windows/apps). The app is written in C++ /CX
+and provides a simple wrapper around the OpenThread API, hidding the raw C/C++ interface.
```
Comment (jwhui) [inline examples/apps/windows/README.md]: typo: `hidding` -> `hiding`

**Comment by jwhui** [comment]
FWIW, `docs/images` was originally used for images needed to support doxygen output.  It might make more sense to locate the windows-specific images closer to the docs where they are actually included/used.

**Comment by nibanks** [comment]
I used that location because other readme files had their images there already. I don't have a better location, and personally feel that a centralized location would be best. Do you have a specific place you would want me to move these images?

**Comment by jwhui** [comment]
I think the current location is fine.  Wasn't sure if you preferred to have the images in a more localized place.

**Pull request merged**